src/kiutils/symbol.py

```diff
@@ -189,10 +190,60 @@ class Symbol():
         https://dev-docs.kicad.org/en/file-formats/sexpr-intro/index.html#_symbols
     """
 
-    id: str = ""
     """Each symbol must have a unique "LIBRARY_ID" for each top level symbol in the library or a unique
     "UNIT_ID" for each unit embedded in a parent symbol. Library identifiers are only valid it top
     level symbols and unit identifiers are on valid as unit symbols inside a parent symbol."""
+    @property
+    def id(self):
+        unit_style_ids = f"_{self.unitId}_{self.styleId}" if (self.unitId is not None and self.styleId is not None) else ""
+        if self.libraryNickname:
+            return f'{self.libraryNickname}:{self.entryName}{unit_style_ids}'
+        else:
+            return f'{self.entryName}{unit_style_ids}'
+
+    @id.setter
+    def id(self, symbol_id):
+        # Split library id into nickname, entry name, unit id and style id (if any)
+        parse_symbol_id = re.match(r"^(.+?):(.+?)_(\d+?)_(\d+?)$", symbol_id)
+        if parse_symbol_id:
+            self.libraryNickname = parse_symbol_id.group(1)
+            self.entryName = parse_symbol_id.group(2)
+            self.unitId = int(parse_symbol_id.group(3))
+            self.styleId = int(parse_symbol_id.group(4))
+        else:
+            parse_symbol_id = re.match(r"^(.+?):(.+?)$", symbol_id)
+            if parse_symbol_id:
+                self.libraryNickname = parse_symbol_id.group(1)
+                entryName_t = parse_symbol_id.group(2)
+            else:
+                entryName_t = symbol_id
+
+            parse_symbol_id = re.match(r"^(.+?)_(\d+?)_(\d+?)$", entryName_t)
+            if parse_symbol_id:
+                self.entryName = parse_symbol_id.group(1)
+                self.unitId = int(parse_symbol_id.group(2))
+                self.styleId = int(parse_symbol_id.group(3))
+            else:
+                if self.libraryNickname:
+                    self.entryName = entryName_t
+                else:
+                    self.entryName = symbol_id
+
+        # Update Value property
+        for property in self.properties:
+            if property.key == 'Value':
+                property.value = self.entryName
+        
+        # Update units id to match parent id
+        for unit in self.units:
+            unit.id = self.id
```

Comment:
Only the `entryName` needs to be propagated to a top-level symbol's child units. Changing this as follows should do the trick here:
```
for unit in self.units:
  unit.entryName = self.entryName
```

Example here is from a symbol list inside a schematic:
```
  (lib_symbols
    (symbol "Switch:SW_Coded" (in_bom yes) (on_board yes)
      // top-level symbol stuff ...
      (symbol "SW_Coded_0_1"
        // child symbol stuff
      )
      (symbol "SW_Coded_1_1"
        // child symbol stuff
      )
    )
  )
```

The `entryName` of the top level symbol is "SW_Coded", which is also present in its child symbols. Changing the ID of the top-level symbol should propagate this down its child symbols. 

For child symbols, there are no further subsymbols to propagate, so this change should work there too.

When a symbol is inside a symbol library, only the part prepending the colon (:) is omitted for the top level symbol, the rest stays the same as in the example. So it should work there too as well.

Reply:
Makes sense, oversight on my side sorry!